ui/src/components/layout/FlowCanvas.tsx

```diff
@@ -87,9 +88,25 @@ const FlowCanvas: React.FC<Props> = () => {
   }
 
   const onNodesChange = useCallback(
-    (changes) => {
+    async (changes) => {
       setNodes((nds) => applyNodeChanges(changes, nds))
-      /**
+      const store = getDataStore();
+      if (changes[0].type === 'remove') {
+        // use its ID to get a handle on it
+        const planId = store.getCurrentPlanId();
+        const displayNodes: Array<DisplayNode> = store.getDisplayNodes(planId);
+        const nodeToDelete = displayNodes.find((obj) => obj.id === changes[0].id);
```

Comment:
The last two lines are not necessary, we just need to call:
```const nodeToDelete = store.getCursor(DisplayNode.getPath(planId, changes[0].id));```
That goes directly to the object itself.
I'm updating this in my rebase.